catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/MetacardIdEqualityFilterDelegate.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.source.solr;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.filter.impl.SimpleFilterDelegate;
+
+/**
+ * Filter delegate that returns all IDs used in metacard ID equality filters.
+ */
+public class MetacardIdEqualityFilterDelegate extends SimpleFilterDelegate<Set<String>> {
```

Comment:
This was designed to really only address the security check when an update/delete is done immediately after a create/update.  But I think you are right that this might cause issues for custom queries for IDs that include `not`s that are done immediately after a create/update.  I will add in some guards to be safe.